airbyte-cdk/python/airbyte_cdk/sources/declarative/extractors/record_filter.py

```diff
@@ -36,3 +37,72 @@ def filter_records(
         for record in records:
             if self._filter_interpolator.eval(self.config, record=record, **kwargs):
                 yield record
+
+
+class ClientSideIncrementalRecordFilterDecorator:
+    """
+    Applies a filter to a list of records to exclude those that are older than the stream_state/start_date.
+
+    :param DatetimeBasedCursor date_time_based_cursor: Cursor used to extract datetime values
+    :param RecordFilter record_filter: Optional filter to apply after date-time filtering
+    :param PerPartitionCursor per_partition_cursor: Optional Cursor used for mapping cursor value in nested stream_state
+
+    """
+
+    def __init__(
+        self,
+        date_time_based_cursor: DatetimeBasedCursor,
+        record_filter: Optional[RecordFilter],
+        per_partition_cursor: Optional[PerPartitionCursor] = None,
+    ):
+        self._date_time_based_cursor = date_time_based_cursor
+        self._delegate = record_filter
+        self._per_partition_cursor = per_partition_cursor
+
+    @property
+    def _cursor_field(self) -> Union[str, Any]:
+        return self._date_time_based_cursor._cursor_field.eval(self._date_time_based_cursor.config)
```

Comment:
let's make`cursor_field` public instead of accessing the private field. I think it's a reasonable expectation for `DatetimeBasedCursor` to have a `cursor_field`

Reply:
agreed, updated

Reply:
@girarda , after change of `cursor_field` -> `cursor_field` in https://github.com/airbytehq/airbyte/pull/38099/commits/c9b5e41031f01a47a1823be0998c9d17af940565

mypy raises an exception:

```
datetime_based_cursor.py:185: error: Item "str" of "InterpolatedString | str" has no attribute "eval"  [union-attr]
```

It seems `mypy` does not take into account `post_init` changes,
Should we just ignore this ?

Reply:
you can add a `# type: ignore  # cursor_field is converted to an InterpolatedString in __post_init__`

Reply:
fixed